.ci/magician/cmd/generate_comment.go

```diff
@@ -265,58 +265,68 @@ func execGenerateComment(prNumber int, ghTokenMagicModules, buildId, buildStep,
 		if err != nil {
-			fmt.Println("computing changed schema labels: ", err)
-			errors[repo.Title] = append(errors[repo.Title], "The diff processor crashed while computing changed schema labels.")
+			fmt.Println("computing changed resource schemas: ", err)
+			errors[repo.Title] = append(errors[repo.Title], "The diff processor crashed while computing changed resource schemas.")
 		}
-		for _, serviceLabel := range serviceLabels {
-			uniqueServiceLabels[serviceLabel] = struct{}{}
+		for _, resource := range affectedResources {
+			uniqueAffectedResources[resource] = struct{}{}
 		}
 	}
 	breakingChangesSlice := maps.Keys(uniqueBreakingChanges)
 	sort.Strings(breakingChangesSlice)
 	data.BreakingChanges = breakingChangesSlice
 
 	// Compute affected resources based on changed files
-	affectedResources := map[string]struct{}{}
+	changedFilesAffectedResources := map[string]struct{}{}
 	for _, repo := range []source.Repo{tpgRepo, tpgbRepo} {
 		if !repo.Cloned {
 			fmt.Println("Skipping changed file service labels; repo failed to clone: ", repo.Name)
 			continue
 		}
 		for _, path := range repo.ChangedFiles {
 			if r := fileToResource(path); r != "" {
-				affectedResources[r] = struct{}{}
+				uniqueAffectedResources[r] = struct{}{}
+				changedFilesAffectedResources[r] = struct{}{}
 			}
 		}
 	}
-	fmt.Printf("affected resources based on changed files: %v\n", maps.Keys(affectedResources))
+	fmt.Printf("affected resources based on changed files: %v\n", maps.Keys(changedFilesAffectedResources))
 
-	// Compute additional service labels based on affected resources
+	// Compute service labels based on affected resources
+	uniqueServiceLabels := map[string]struct{}{}
 	regexpLabels, err := labeler.BuildRegexLabels(labeler.EnrolledTeamsYaml)
 	if err != nil {
 		fmt.Println("error building regexp labels: ", err)
 		errors["Other"] = append(errors["Other"], "Failed to parse service label mapping")
 	}
 	if len(regexpLabels) > 0 {
-		for _, label := range labeler.ComputeLabels(maps.Keys(affectedResources), regexpLabels) {
+		for _, label := range labeler.ComputeLabels(maps.Keys(uniqueAffectedResources), regexpLabels) {
 			uniqueServiceLabels[label] = struct{}{}
 		}
 	}
 
-	// Add service labels to PR
+	// Add service labels to PR if it doesn't already have service labels
 	if len(uniqueServiceLabels) > 0 {
-		serviceLabelsSlice := maps.Keys(uniqueServiceLabels)
-		sort.Strings(serviceLabelsSlice)
-		if err = gh.AddLabels(strconv.Itoa(prNumber), serviceLabelsSlice); err != nil {
-			fmt.Printf("Error posting new service labels %q: %s", serviceLabelsSlice, err)
-			errors["Other"] = append(errors["Other"], "Failed to update service labels")
+		// short-circuit if service labels have already been added to the PR
+		hasServiceLabels := false
+		for _, label := range pullRequest.Labels {
+			if strings.HasPrefix(label.Name, "service/") {
+				hasServiceLabels = true
+			}
+		}
+		if !hasServiceLabels {
```

Comment:
I might be misreading it. Does it prevent of adding new service labels if new affected resources are detected (potentially result in new service labels are needed) in any following commits of PRs

Reply:
yes - we only add service labels if there aren't already service labels. The idea is that if something is incorrectly identified, we'd want to be able to manually set the service labels and not have them changed by automation. (This is the intended behavior, but it could be that we want to change it in the future if it ends up causing problems.)